apicast/src/configuration/service.lua

```diff
@@ -137,12 +137,24 @@ function backend_version_credentials.version_oauth(config)
   -- Resource servers MUST support this method. [Bearer]
   access_token = access_token or authorization.token
 
-  ------
-  -- oauth credentials.
-  -- @field 1 Access Token
-  -- @field access_token Access Token
-  -- @table credentials_oauth
-  return { access_token, access_token = access_token }
+  if keycloak.configured then
```

Comment:
What I meant in my comment was that this does not have to be in the service object but in the oauth object.

Simply:

```
if backend_version == 'oauth' then
  local o = oauth.new(service)
  credentials = o:credentials(credentials)
end
```

and the normal oauth would just return the object passed in and  keycloak version would extract the app_id from jwt. 

Using polymorphism instead of conditionals.